cql3/functions/aggregate_fcts.cc

```diff
@@ -481,8 +481,15 @@ static std::vector<data_type> state_arg_types_to_uda_arg_types(const std::vector
     return types;
 }
 
+static data_type uda_return_type(::shared_ptr<scalar_function> ffunc, ::shared_ptr<scalar_function> sfunc) {
+    if (ffunc) {
```

Comment:
nitpick: that's also a nice candidate for the ternary operator, as you did in `compute()`:
```cpp
return (ffunc) ? ffunc->return_type() : sfunc->return_type();
```